### PR TITLE
Playable URL capability, tested.

### PR DESCRIPTION
**Pull request by kixxauth**
modified:   lib/fetch-backlot-asset.js
	new file:   lib/fetch-playable-url.js
	modified:   spec/asset-handler-spec.js

### CODE REVIEW

File: lib/fetch-playable-url.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		throw new Error('args.embedCode is required');
+	}
+
+	const params = {
+		method: 'GET',
+		url: `${BASE_URL}/${apiKey}/${embedCode}?${QUERY}`
+	};
+
+	return fetchPlayableUrl.makeRequest(params).then(data => {
+		const stream = (data.authorization_data[embedCode].streams || [])[0];
```
Comment (aljachimiak) [inline lib/fetch-playable-url.js]: You may want to test what happens here when `authorization_data` is absent.  There is a possibility of errors coming back from this endpoint with a statusCode of 200.

File: lib/fetch-playable-url.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+				`Stream undefined for playable URL with apiKey ${apiKey} and embedCode ${embedCode}.`
+			);
+			error.code = 'STREAM_UNDEFINED';
+			return Promise.reject(error);
+		}
+
+		const buff = new Buffer(stream.url.data, 'base64');
+		const deCoded = buff.toString();
+
+		return deCoded.replace(/player\/iphone/, 'player/appletv');
```
Comment (aljachimiak) [inline lib/fetch-playable-url.js:45]: FYI - this was a risky move when I implemented it.   As far as I can tell it has been tested very little if at all on rook devices.

I've informed Ooyala that this is a fix that should probably be in place upstream at the api level.

**Review by aljachimiak — COMMENTED** [review]
Here's a few comments from issues I've dealt with over the past month.

File: lib/fetch-playable-url.js (hunk 1 above)
Comment (kixxauth) [reply lib/fetch-playable-url.js]: thanks @aljachimiak

**Pull request merged**